DOCS-633 Write article for creating basic NFS share. (#124)
freenas/documentation: content/en/docs/sharing/nfs/nfs-share.md

---
title: "Configuring a NFS Share"

> description: "A how-to guide on how to create a general purpose NFS share."

Author: ```suggestion
description: "A how-to guide on creating a general purpose NFS share."
```

---

Creating a Network File System (NFS) share on TrueNAS gives the benefit

> of making lots of data easily accessible for anyone with the information
> to the share. Depending on how the share is setup, users accessing the

Reviewer: ```suggestion
of making lots of data easily available for anyone with share access.
Depending how the share is setup, users accessing the
```

share can be restricted to read or write privileges.

To get started, make sure a
<a href="/docs/initial-setup/storage/datasets">dataset has been created</a>.

> This dataset serves as the storage for the data that is going to be
> shared. If a dataset already exists, proceed to turning the NFS service

Reviewer: ```suggestion
This dataset serves as share data storage.
If a dataset already exists, proceed to turning the NFS service
```

on.

## NFS Service

To turn the NFS service on, go to **Services** and click the slider for
*NFS*. If you wish to turn the service on automatically when the TrueNAS
system is rebooted, check the *Start Automatically* box.

> NOTE: The NFS share will not work if the service is not turned on.

The NFS service settings can be configured by clicking
<i class="fas fa-pen"></i>. Don't forget to click *SAVE* when changing
the settings. Unless a specific setting is needed, it is recommended to
use the default settings for the NFS service.

## NFS Share

Now it is time to create the NFS share. Go to

> **Sharing > Unix Shares (NFS)** and click *ADD*. Use the browser to

Reviewer: ```suggestion
**Sharing > Unix Shares (NFS)** and click *ADD*. Use the file browser to
```

select the dataset to be shared. An optional *Description* can be set to
help identify the share. At the time of creation, the NFS share is

> enabled by default. If you wish to create the share but not enable it
> yet, do not check the *Enable* checkbox. Click *SUBMIT*, to create the
> share.

Reviewer: ```suggestion
enabled by default. If you wish to create the share but not immediately enable it,
unset the *Enable* checkbox. Clicking *SUBMIT* creates the share.
```

> To configure more advanced options for the share, click
> *ADVANCED OPTIONS*.

Reviewer: ```suggestion
Opening the *ADVANCED OPTIONS* allows tuning the share access
permissions and defining authorized networks.
```

Existing NFS shares can be edited by going to
**Sharing > Unix Shares (NFS)** and clicking
<i class="fas fa-ellipsis-v"></i>.

## Connecting to the NFS Share

Although you can connect to an NFS share with various operating systems,
it is recommended to use a Linux/Unix operating system. First, download
the *nfs-common* kernel module. This can be done using the package

> manager of the installed distribution. For example, on Ubuntu/Debian

Reviewer: ```suggestion
manager of the installed distribution. For example, on Ubuntu/Debian,
```
